contracts/ERC1155PresetPausableNonTransferrable.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.13;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/presets/ERC1155PresetMinterPauserUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/extensions/ERC1155SupplyUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC777/IERC777RecipientUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC777/ERC777Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC777/IERC777RecipientUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC1820ImplementerUpgradeable.sol";
+
+import "hardhat/console.sol"; // todo rm
+
+/**
+ * @title ERC1155PresetPausableNonTransferrable
+ */
+contract ERC1155PresetPausableNonTransferrable is
+  ERC1155SupplyUpgradeable,
+  ERC1155PresetMinterPauserUpgradeable
+{
+  error ForbiddenTransferAfterMinting();
+
+  /**
+   * @notice Role conferring operator permissions
+   *
+   * @dev This role is assigned to operators which can transfer certificates from an address to another by bypassing
+   * the `_beforeTokenTransfer` hook.
+   */
+  bytes32 public constant CERTIFICATE_OPERATOR_ROLE =
+    keccak256("CERTIFICATE_OPERATOR_ROLE");
+
+  // slither-disable-next-line naming-convention, solhint-disable-next-line func-name-mixedcase
+  function __ERC1155PresetPausableNonTransferrable_init_unchained()
+    internal
+    onlyInitializing
+  {
+    _grantRole(CERTIFICATE_OPERATOR_ROLE, _msgSender());
+  }
+
+  // slither-disable-next-line naming-convention, solhint-disable-next-line func-name-mixedcase
+  function __ERC1155PresetPausableNonTransferrable_init(string memory uri_)
+    internal
+    onlyInitializing
+  {
+    // todo verify all inherited initializers are called
```

Comment:
did a little reading on `init` v.s. `init_unchained` and from what I understand, if we're using the unchained initializers we just need to make sure that our contract calls the unchained initializer for each unique parent contract in the inheritance tree?